DNN Platform/Library/Services/GeneratedImage/ImageHandlerInternal.cs

```diff
@@ -231,6 +233,45 @@ public void HandleImageRequest(HttpContextBase context, Func<NameValueCollection
 
             string cacheId = GetUniqueIDString(context, uniqueIdStringSeed);
 
+            var hasprofileChanged = false;
```

Comment:
This is incorrect way to check the changes and avoid the cache. In fact the cache should be cleared by subscribing to OnProfileUpdated event of ProfileUserControlBase. Further, I don't see any changes where Current user login is being verified or the profile access level is checked.